src/player.cpp

```diff
@@ -4902,11 +4913,11 @@ dealt_damage_instance player::deal_damage(Creature* source, body_part bp, const
             add_msg_player_or_npc(m_bad, _("%s grabs you!"), _("%s grabs <npcname>!"),
                                   source->disp_name().c_str());
             if (has_grab_break_tec() && get_grab_resist() > 0 && get_dex() > get_str() ?
-                dice(get_dex(), 10) : dice(get_str(), 10) > dice(source->get_dex(), 10)) {
+                dice(get_dex(), 10) : dice(get_str(), 10) > dice(8, 10)) {
                 add_msg_player_or_npc(m_good, _("You break the grab!"),
                                       _("<npcname> breaks the grab!"));
             } else {
-                add_effect("grabbed", 1);
+                add_effect("grabbed", 2);
```

Comment:
Yea, occasionally the grabbed effect wouldn't apply from one turn. I never noticed any weird moments of being far away and still being grabbed.